File: Hero/Models/BlastZone/PropertyLimitBlastZoneCapacityRequest.cs
namespace Hero.Models.BlastZone;

public sealed class PropertyLimitBlastZoneCapacityRequest
{
    public DateTime InceptionDate { get; init; }

    public DateTime ExpiryDate { get; init; }

    public DateTime? ReservationExpiryDate { get; init; }

    public List<PropertyLimit> PropertyLimits { get; init; }

    public long FloatingValue { get; init; }

    public long? FirstLossLimitValue { get; init; }

    public Guid? ClientId { get; init; }

    public bool IsRenewable { get; init; }

    public Guid? OriginalGroupId { get; init; }

    public int BinderSectionId { get; init; }

    public string QuoteCurrencyIsoCode { get; init; }
}

File: Hero/ClientApp/src/app/models/property-limit-blast-zone-capacity-request.ts
import { PropertyLimit } from '@app/quote/models/property-limit.model';

export class PropertyLimitBlastZoneCapacityRequest {
    public propertyLimits: PropertyLimit[];
    public inceptionDate: Date;
    public expiryDate: Date;
    public reservationExpiryDate?: Date;
    public firstLossLimitValue?: number;
    public floatingValue?: number;
    public clientId?: string;
    public isRenewable?: boolean;
    public originalGroupId?: string;
    public binderSectionId?: number;
    public quoteCurrencyIsoCode?: string;
}
